Online/RPCs.cs

```diff
@@ -121,49 +121,86 @@ public static void GoToDeathScreen()
+                game.GetStorySession.saveState.denPosition = denPos;
+                (OnlineManager.lobby.gameMode as StoryGameMode).defaultDenPos = denPos;
+            }
             foreach (OnlinePlayer player in OnlineManager.players)
             {
-                if (!player.OutgoingEvents.Any(e => e is RPCEvent rpc && rpc.IsIdentical(RPCs.GoToWinScreen, malnurished)))
+                if (!player.OutgoingEvents.Any(e => e is RPCEvent rpc && rpc.IsIdentical(RPCs.GoToWinScreen, malnourished, denPos)))
                 {
-                    player.InvokeRPC(RPCs.GoToWinScreen, malnurished);
+                    if (player.isMe)
+                    {
+                        GoToWinScreen(malnourished, denPos);
+                    }
+                    else
+                    {
+                        player.InvokeRPC(RPCs.GoToWinScreen, malnourished, denPos);
+                    }
                 }
             }
         }
 
         //Assumed to be called for storymode only
         [RPCMethod]
-        public static void GoToWinScreen(bool malnourished)
+        public static void GoToWinScreen(bool malnourished, string denPos)
         {
+            RainMeadow.Debug($"({malnourished})");
             var game = (RWCustom.Custom.rainWorld.processManager.currentMainLoop as RainWorldGame);
-            if (game == null || game.manager.upcomingProcess != null)
+
+            RainMeadow.Debug($"RainWorldGame = {RWCustom.Custom.rainWorld.processManager.currentMainLoop}");
+
+            if (game == null)
             {
+                RainMeadow.Debug($"game is null");
                 return;
             }
 
-            if (!malnourished && !game.rainWorld.saveBackedUp)
+            if (game.manager.upcomingProcess != null)
             {
-                game.rainWorld.saveBackedUp = true;
-                game.rainWorld.progression.BackUpSave("_Backup");
-            }
-
-            //This needs to be called after shelterDoor::Close and game state synced for this to be accurate.
-            var denPos = (OnlineManager.lobby.gameMode as StoryGameMode).defaultDenPos;
-            if (denPos == null) {
-                AbstractCreature firstAlivePlayer = game.FirstAlivePlayer;
-                denPos = game.world.GetAbstractRoom(firstAlivePlayer.pos).name;
+                RainMeadow.Debug($"game has upcoming process");
+                return;
             }
 
-            //TODO: Having soft-win on makes this very difficult. For now I'm (Turtle) locking it down so that you can only win if the host is alive
             if (OnlineManager.lobby.isOwner) {
+                if (!malnourished && !game.rainWorld.saveBackedUp)
+                {
+                    game.rainWorld.saveBackedUp = true;
+                    game.rainWorld.progression.BackUpSave("_Backup");
+                }
                 game.GetStorySession.saveState.SessionEnded(game, true, malnourished);
```

Comment:
So this guy is the thing you'll want to "sync". The food pips graphical error occurs because client's don't run this segment of code. So when a client does "go to win screen" it doesn't do the whole cycle end properly

Reply:
can we get away with hooking the save functions to not write to disk?

Reply:
Short answer, no. 

Longer answer, the issue comes down to a null check the game itself makes. Local co-op gets around this by making all players have to hibernate in the same area. So they can just assume the hibernation shelter is X and continue with the saving logic